// @framer/fhur.got-bootstrap/code/alert.tsx
import * as React from "react";
import { PropertyControls, ControlType } from "framer";

const style: React.CSSProperties = {
  height: "100%"
};

// Define type of property
interface Props {
  text: string;
  alertType: string;
}

export class alert extends React.Component<Props> {
  // Set default properties
  static defaultProps = {
    text: "Hello World!",
    alertType: "alert-primary",
    height: 50
  };

  // Items shown in property panel
  static propertyControls: PropertyControls = {
    text: { type: ControlType.String, title: "Text" },
    alertType: {
      type: ControlType.Enum,
      title: "Alert Type",
      options: [
        "alert-primary",
        "alert-secondary",
        "alert-success",
        "alert-danger",
        "alert-warning",
        "alert-info",
        "alert-light",
        "alert-dark"
      ]
    }
  };

  render() {
    return (
      <div
        className={`alert ${this.props.alertType}`}
        role="alert"
        style={style}
      >
        {this.props.text}
      </div>
    );
  }
}
