// @framer/fhur.got-bootstrap/code/breadcrumbs.tsx
import * as React from "react";
import { PropertyControls, ControlType } from "framer";

const style: React.CSSProperties = {
  height: "100%",
  display: "flex",
  alignItems: "center",
  justifyContent: "center",
  textAlign: "center",
  color: "#8855FF",
  background: "rgba(136, 85, 255, 0.1)",
  overflow: "hidden"
};

// Define type of property
interface Props {
  content: string;
}

const contentToBreadcrumbItems = (content: string) => {
  const elems = content.split(",").map(x => x.trim());

  return elems.map((el, index) => {
    if (index === elems.length - 1) {
      return (
        <li className="breadcrumb-item active" aria-current="page">
          {el}
        </li>
      );
    }
    return (
      <li className="breadcrumb-item">
        <a href="#">{el}</a>
      </li>
    );
  });
};

export class breadcrumbs extends React.Component<Props> {
  // Set default properties
  static defaultProps = {
    content: "Home, Library, Data",
    height: 50,
    width: 250
  };

  // Items shown in property panel
  static propertyControls: PropertyControls = {
    content: { type: ControlType.String, title: "Comma Separated Breadcrumbs" }
  };

  render() {
    return (
      <nav aria-label="breadcrumb">
        <ol className="breadcrumb">
          {contentToBreadcrumbItems(this.props.content)}
        </ol>
      </nav>
    );
  }
}
