// @framer/fhur.got-bootstrap/code/button.tsx
import * as React from "react";
import { PropertyControls, ControlType } from "framer";

const style: React.CSSProperties = {
  width: "100%"
};

// Define type of property
interface Props {
  text: string;
  buttonType: string;
  buttonSize: string;
  enabled: boolean;
}

export class button extends React.Component<Props> {
  // Set default properties
  static defaultProps = {
    text: "Hello World!",
    buttonType: "btn-primary",
    buttonSize: "",
    enabled: true,
    height: 40,
    width: 115
  };

  // Items shown in property panel
  static propertyControls: PropertyControls = {
    text: { type: ControlType.String, title: "Text" },
    buttonType: {
      type: ControlType.Enum,
      title: "Type",
      options: [
        "btn-primary",
        "btn-secondary",
        "btn-success",
        "btn-danger",
        "btn-warning",
        "btn-info",
        "btn-light",
        "btn-dark",
        "btn-link"
      ]
    },
    buttonSize: {
      type: ControlType.Enum,
      title: "Size",
      options: ["unknown", "btn-sm", "btn-lg"],
      optionTitles: ["Regular", "btn-sm", "btn-lg"]
    },
    enabled: {
      type: ControlType.Boolean,
      title: "Enabled"
    }
  };

  render() {
    return (
      <button
        type="button"
        className={`btn ${this.props.buttonType} ${this.props.buttonSize}`}
        disabled={!this.props.enabled}
        style={style}
      >
        {this.props.text}
      </button>
    );
  }
}
