// @framer/fhur.got-bootstrap/code/input_group.tsx
import * as React from "react";
import { PropertyControls, ControlType } from "framer";

const style: React.CSSProperties = {};

// Define type of property
interface Props {
  placeholder: string;
  groupPrepend: string;
  groupAppend: string;
  inputType: string;
}

export class input_group extends React.Component<Props> {
  // Set default properties
  static defaultProps = {
    placeholder: "99",
    groupPrepend: "$",
    groupAppend: ".00",
    inputType: "string",
    height: 40
  };

  // Items shown in property panel
  static propertyControls: PropertyControls = {
    placeholder: { type: ControlType.String, title: "Text" },
    groupPrepend: { type: ControlType.String, title: "Text" },
    groupAppend: { type: ControlType.String, title: "Text" },
    inputType: { type: ControlType.String, title: "Text" }
  };

  render() {
    return (
      <div className="input-group" style={style}>
        {this.props.groupPrepend.length > 0 && (
          <div className="input-group-prepend">
            <span className="input-group-text">{this.props.groupPrepend}</span>
          </div>
        )}

        <input
          type={this.props.inputType}
          className="form-control"
          placeholder={this.props.placeholder}
          aria-label={this.props.placeholder}
        />

        {this.props.groupAppend.length > 0 && (
          <div className="input-group-append">
            <span className="input-group-text">{this.props.groupAppend}</span>
          </div>
        )}
      </div>
    );
  }
}
