// @framer/fhur.got-bootstrap/code/nav.tsx
import * as React from "react";
import { PropertyControls, ControlType } from "framer";

const style: React.CSSProperties = {
  height: "100%"
};

// Define type of property
interface Props {
  navType: string;
  content: string;
  activeIndex: number;
}

const contentAsNavItems = (content: string, activeIndex: number) => {
  return content
    .split(",")
    .map(x => x.trim())
    .map((elem, index) => {
      return (
        <li className="nav-item">
          <a
            className={`nav-link ${activeIndex === index ? "active" : ""}`}
            href="#"
          >
            {elem}
          </a>
        </li>
      );
    });
};

export class nav extends React.Component<Props> {
  // Set default properties
  static defaultProps = {
    navType: "nav-pills",
    content: "First Link, Second Link, Third Link",
    activeIndex: 2,
    width: 326,
    height: 40
  };

  // Items shown in property panel
  static propertyControls: PropertyControls = {
    navType: {
      type: ControlType.Enum,
      title: "Nav Type",
      options: ["nav-pills", "nav-tabs", "unknown"]
    },
    content: {
      type: ControlType.String,
      title: "Comma Separated Nav Items"
    },
    activeIndex: {
      type: ControlType.Number,
      min: 0,
      title: "Active Index"
    }
  };

  render() {
    return (
      <ul
        className={`nav ${
          this.props.navType === "unknown" ? "" : this.props.navType
        }`}
        style={style}
      >
        {contentAsNavItems(this.props.content, this.props.activeIndex)}
      </ul>
    );
  }
}
