// @framer/fhur.got-bootstrap/code/pagination.tsx
import * as React from "react";
import { PropertyControls, ControlType } from "framer";

const style: React.CSSProperties = {
  width: "100%"
};

// Define type of property
interface Props {
  textPrevious: string;
  textNext: string;
  pages: string;
}

const pagesStringToComponents = (pages: string) => {
  return pages
    .split(",")
    .map(str => str.trim())
    .map(pageName => {
      return (
        <li key={pageName} className="page-item">
          <a className="page-link" href="#">
            {pageName}
          </a>
        </li>
      );
    });
};

export class pagination extends React.Component<Props> {
  // Set default properties
  static defaultProps = {
    textPrevious: "Previous",
    textNext: "Next",
    pages: "1,2,3,4",
    height: 40,
    width: 285
  };

  // Items shown in property panel
  static propertyControls: PropertyControls = {
    textPrevious: { type: ControlType.String, title: "Text Next" },
    textNext: { type: ControlType.String, title: "Text Previous" },
    pages: { type: ControlType.String, title: "Comma Separated Pages" }
  };

  render() {
    return (
      <nav aria-label="Page Navigation" style={style}>
        <ul className="pagination">
          <li className="page-item">
            <a className="page-link" href="#">
              {this.props.textPrevious}
            </a>
          </li>
          {pagesStringToComponents(this.props.pages)}
          <li className="page-item">
            <a className="page-link" href="#">
              {this.props.textNext}
            </a>
          </li>
        </ul>
      </nav>
    );
  }
}
