// @framer/fhur.got-bootstrap/code/progress.tsx
import * as React from "react";
import { PropertyControls, ControlType } from "framer";

const style: React.CSSProperties = {
  height: "100%"
};

// Define type of property
interface Props {
  percentage: number;
  progressType: string;
  label: boolean;
  striped: string;
}

export class progress extends React.Component<Props> {
  // Set default properties
  static defaultProps = {
    percentage: 25,
    progressType: "bg-success",
    label: true,
    striped: "striped-and-animated",
    height: 18
  };

  // Items shown in property panel
  static propertyControls: PropertyControls = {
    percentage: {
      type: ControlType.Number,
      title: "Percentage",
      min: 0,
      max: 100
    },
    progressType: {
      type: ControlType.Enum,
      title: "Progress Type",
      options: ["bg-success", "bg-info", "bg-warning", "bg-danger"]
    },
    striped: {
      type: ControlType.Enum,
      title: "Background",
      options: [
        "unknown",
        "progress-bar-striped",
        "progress-bar-striped progress-bar-animated"
      ],
      optionTitles: ["Regular", "Striped", "Striped and Animated"]
    },
    label: {
      type: ControlType.Boolean,
      title: "Label"
    }
  };

  render() {
    return (
      <div className="progress" style={style}>
        <div
          className={`progress-bar ${this.props.progressType} ${
            this.props.striped
          }`}
          role="progressbar"
          style={{ width: `${this.props.percentage}%` }}
          aria-valuenow={this.props.percentage}
          aria-valuemin={0}
          aria-valuemax={100}
        >
          {this.props.label ? `${this.props.percentage}%` : ""}
        </div>
      </div>
    );
  }
}
